src/dataset.jl

```diff
@@ -510,23 +510,45 @@ filelist(dataset::AbstractDataset)::Vector{String} =
 
 Fetch the layer at index `i` (between `0` and `nlayer(dataset)-1`)
 
-The returned layer remains owned by the `Dataset` and should not be deleted by
+The returned layer remains owned by the `dataset` and should not be deleted by
 the application.
 """
 getlayer(dataset::AbstractDataset, i::Integer)::IFeatureLayer =
     IFeatureLayer(GDAL.gdaldatasetgetlayer(dataset.ptr, i), ownedby = dataset)
 
+"""
+    getlayer(dataset::AbstractDataset)
+
+Fetch the first layer and raise an error if `dataset` contains more than one layer
+
+The returned layer remains owned by the `dataset` and should not be deleted by
+the application.
+"""
+function getlayer(dataset::AbstractDataset)::IFeatureLayer
+    layer = IFeatureLayer(
+        GDAL.gdaldatasetgetlayer(dataset.ptr, 0),
+        ownedby = dataset,
+    )
+    nlayer(dataset) == 1 ||
+        error("Dataset has multiple layers. Specify the layer number or name")
```

Comment:
nit/optional: I'd move this check before the previous line so that we (i) check for the existence of a layer before calling `GDAL.gdaldatasetgetlayer(dataset.ptr, 0)`, (ii) exit earlier in the event of an error.

Same below.

Reply:
Thanks for pointing this out.
I have also modified `unsafe_getlayer` in the same fashion

Reply:
Resolved by https://github.com/yeesian/ArchGDAL.jl/pull/257/commits/30ff7510836cc6841af83ed4373b48f09ec1acbc